### PR TITLE
Application confirmation

### PR DESCRIPTION
**Pull request by ofpau**
Solves #8

### CODE REVIEW

File: register/templates/confirm.html [hunk 1]
```diff
@@ -1,21 +1,44 @@
...
+            {% else %}
+
+            <div class="panel panel-success">
+                <div class="panel-heading">
+                    <h3 class="panel-title">Thanks for confirming, {{ application.name }}!</h3>
+                </div>
+                <div class="panel-body">
+
+                    {% if reconfirming %}
+                        You already confirmed your spot before, {{ application.name }}. But we love your enthusiasm :)
```
Comment (casassg) [inline register/templates/confirm.html:32]: :D

File: register/models.py [hunk 2]
```diff
@@ -76,9 +76,13 @@ def invite(self, request):
...
 
-    def confirm(self):
-        if self.status != 'I':
+    def is_confirmed(self):
+        return self.status == 'C'
+
+    def confirm(self, cancellation_url):
+        if self.status != 'I' and self.status != 'C':
             raise ValidationError('Application hasn\'t been invited yet')
+        self._send_confirmation_ack(cancellation_url)
```
Comment (casassg) [inline register/models.py]: Send only once, only if accepted for the first time.  Same for the status, only update application if it's not confirmed already (no need to do unnecessary updates to the DB)

File: register/templates/confirm.html [hunk 3]
```diff
@@ -1,21 +1,44 @@
...
+                    {% else %}
+                        {{ application.name }}, you just confirmed your spot at HackUPC Spring 2017!
+                    {% endif %}
+                    We can't wait to have you here.
+
+                </div>
             </div>
+            {% endif %}
+            If you can't make it anymore, you can still cancel your participation <a href="{{ cancel }}">here</a>.
```
Comment (casassg) [inline register/templates/confirm.html:40]: If error this is showed too. Fix to only show when no error.

File: register/templates/confirm.html [hunk 4]
```diff
@@ -1,21 +1,44 @@
...
-                <h3 class="panel-title">Thanks for confirming {{ application.name }}</h3>
+
+    <div class="confirm-content">
+        <div class="col-md-8 col-md-offset-2">
+            {{ self.title }}
+
+            {% if error %}
+            <div class="panel panel-danger">
+                <div class="panel-heading">
+                    <h3 class="panel-title">Hey, {{ application.name }}!</h3>
```
Comment (casassg) [inline register/templates/confirm.html]: Probably better something like: "We are sorry {{application.name}}"

File: register/views.py [hunk 5]
```diff
@@ -19,18 +19,21 @@ def get_context_data(self, **kwargs):
...
-                'cancel': application.cancelation_url(request)
+                'cancel': cancellation_url,
+                'reconfirming': already_confirmed
             })
-        except ValidationError:
+        except ValidationError as e:
             context.update({
-                'error': "application can't be confirmed",
+                'application': application,
+                'error': e.message,
```
Comment (casassg) [inline register/views.py]: Use custom message, replace it ValidationError creation. Something like: "This application hasn't been invited yet" (better readability, than "Application hasn't been invited")

**Review by casassg — CHANGES_REQUESTED** [review]
It looks good, just a few things to solve before merging

**Comment by ofpau** [comment]
@casassg requested changes added!

**Review by casassg — APPROVED** [review]
LGTM, :sheep: it

**Pull request merged**